SampleApplication.py

```diff
@@ -1,40 +1,77 @@
 import AbstractApplication as Base
-from time import sleep
+from threading import Semaphore
+
 
 class SampleApplication(Base.AbstractApplication):
+    # setup our Application
+    def __init__(self):
+        super().__init__()
+
+        # Semaphores for async execution. They make sure that the action is completed.
+        self.gesture_lock = Semaphore(0)
+        self.language_lock = Semaphore(0)
+        self.text_lock = Semaphore(0)
+        self.eyecolor_lock = Semaphore(0)
+        self.name_lock = Semaphore(0)
+
+        # Attributes of our application
+        self.user_name = None
+
     def main(self):
+        # Setting language
         self.setLanguage('en-US')
+        self.language_lock.acquire()
+
+        # Change Eyecolor
+        self.setEyeColour('red')
+        self.eyecolor_lock.acquire()
+
+        # Saying something with body language
         self.sayAnimated('Hello Group 7.')
-        sleep(2)
-        # test comment
+        self.text_lock.acquire()
+
+        # Say something without moving
+        self.say('Boring hello group 7.')
+        self.text_lock.acquire()
+
+        # Do a gesture
+        # untitled-2ae403 is the ID of the Project in Choreographe.
+        # Did not know this in advance, hence the awful name.
+        self.doGesture('untitled-2ae403/behavior_1')
+        self.gesture_lock.acquire()
+
+        # TODO: How to set this up properly
+        # Pass the required Dialogflow parameters (add your Dialogflow parameters)
+        # self.setDialogflowKey('<keyfile>.json')
+        # self.setDialogflowAgent('<agentname>')
+
+        # Listen for an answer for at most 5 seconds
+        # This snipped assumes a DialogueFlow application listening for a name
+        self.setAudioContext('answer_name')
+        self.startListening()
+        self.name_lock.acquire(timeout=5)
+        self.stopListening()
+
+    def onAudioIntent(self, *args, intentName):
```

Comment:
We should probably be consistent with out case and use _snake_case_ for variables where we can

Reply:
Yes. For Python I normaly use the snake_case. The camelCase is all related to the code we got from them. I don't mind which one we do.

Reply:
let's just do camelCase so we are consistent in the whole project

Reply:
you mean snake_case? we use it more often